src/material-experimental/mdc-chips/chip-action.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  AfterViewInit,
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  Inject,
+  OnChanges,
+  OnDestroy,
+  SimpleChanges,
+} from '@angular/core';
+import {DOCUMENT} from '@angular/common';
+import {
+  MDCChipActionAdapter,
+  MDCChipActionFoundation,
+  MDCChipActionType,
+  MDCChipPrimaryActionFoundation,
+} from '@material/chips';
+import {emitCustomEvent} from './emit-event';
+import {
+  CanDisable,
+  HasTabIndex,
+  mixinDisabled,
+  mixinTabIndex,
+} from '@angular/material-experimental/mdc-core';
+import {BooleanInput, NumberInput} from '@angular/cdk/coercion';
+
+const _MatChipActionMixinBase = mixinTabIndex(mixinDisabled(class {}), -1);
+
+/** @docs-private */
+@Directive({
+  selector: '[matChipAction]',
+  inputs: ['disabled', 'tabIndex'],
+  host: {
+    '[attr.tabindex]': 'disabled ? null : tabIndex',
+    '[attr.disabled]': "disabled ? '' : null",
+    '[attr.aria-disabled]': 'disabled',
+    '(click)': '_handleClick()',
+    '(keydown)': '_handleKeydown($event)',
+  },
+})
+export class MatChipAction
+  extends _MatChipActionMixinBase
+  implements AfterViewInit, OnDestroy, CanDisable, HasTabIndex, OnChanges
+{
+  private _document: Document;
+  private _foundation: MDCChipActionFoundation;
+  private _adapter: MDCChipActionAdapter = {
+    focus: () => this.focus(),
+    getAttribute: (name: string) => this._elementRef.nativeElement.getAttribute(name),
+    setAttribute: (name: string, value: string) => {
+      // MDC tries to update the tabindex directly in the DOM when navigating using the keyboard
+      // which overrides our own handling. If we detect such a case, assign it to the same property
+      // as the Angular binding in order to maintain consistency.
+      if (name === 'tabindex') {
+        this._updateTabindex(parseInt(value));
```

Comment:
Should we use `Number(value)` here instead? IIRC `Number` is more predictable (e.g. doesn't interpret a leading `0` as octal)

Reply:
We've been using `parseInt` for cases like this everywhere else so I went along with it. I also think that it's unlikely for somebody to write out `tabindex="007"`.